### PR TITLE
Sitemap support

### LINKED ISSUE #41: Sitemap support
Sylius Sitemap Plugin exists: https://github.com/stefandoorn/sitemap-plugin. Would be great to have a provider to include the page URL's into it. 

Some options:
* Only provide a provider in this repo with tag `sylius.sitemap_provider` and add the sitemap plugin as a suggest to composer.json. And optionally put the activation behind a config setting.
* Provide functionality in a separate repo that requires both this package and the sitemap package + add the provider

@bitbager What do you prefer?

### PR DESCRIPTION
**Pull request by sweoggy**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #41 
| License         | MIT

### CODE REVIEW

**Comment by bitbager** [comment]
Wow! Thank you a lot! 

@patrick477, can I ask you to review it?

File: src/Sitemap/Provider/PageUrlProvider.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+     * @param PageInterface $page
+     *
+     * @return SitemapUrlInterface
+     */
+    private function createProductUrl(PageInterface $page): SitemapUrlInterface
+    {
+        $pageUrl = $this->sitemapUrlFactory->createNew();
+        $pageUrl->setChangeFrequency(ChangeFrequency::daily());
+        $pageUrl->setPriority(0.7);
+        if ($page->getUpdatedAt()) {
```
Comment (stefandoorn) [inline src/Sitemap/Provider/PageUrlProvider.php]: Doesn't `$page` always have an updated at? Else fall back to created at maybe? 

```
$pageUrl->setLastModification($page->getUpdatedAt() ?? $page->getCreatedAt());
```

File: src/Sitemap/Provider/PageUrlProvider.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+            $pageUrl->setLastModification($page->getUpdatedAt());
+        }
+        /** @var PageTranslationInterface $translation */
+        foreach ($this->getTranslations($page) as $translation) {
+            if (!$translation->getLocale()) {
```
Comment (stefandoorn) [inline src/Sitemap/Provider/PageUrlProvider.php]: Is this possible?

File: src/Sitemap/Provider/PageUrlProvider.php [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    {
+        return $this->pageRepository->createQueryBuilder('o')
+            ->addSelect('translation')
+            ->innerJoin('o.translations', 'translation')
+            ->andWhere('o.enabled = :enabled')
+            ->setParameter('enabled', true)
+            ->getQuery()
+            ->getResult();
+    }
+
```
Comment (patrick477) [inline src/Sitemap/Provider/PageUrlProvider.php]: a better way would be to add the `findEnabled` method to the repository

File: tests/Api/Sitemap/Provider/AbstractTestController.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * @var LocaleInterface
+     */
+    protected $locale;
+
+    /**
+     * @var LocaleInterface
+     */
+    protected $locale2;
+
```
Comment (patrick477) [inline tests/Api/Sitemap/Provider/AbstractTestController.php]: a better name would be ` $secondLocale `

File: src/Sitemap/Provider/PageUrlProvider.php [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    /**
+     * @return array
+     */
+    private function getLocaleCodes(): array
+    {
+        if ($this->channelLocaleCodes === null) {
```
Comment (patrick477) [inline src/Sitemap/Provider/PageUrlProvider.php]: a better practice would be ` null === $this->channelLocaleCodes `

File: src/Sitemap/Provider/PageUrlProvider.php [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        }
+        return $this->channelLocaleCodes;
+    }
+
+    /**
+     * @param PageInterface $page
+     *
+     * @return SitemapUrlInterface
+     */
+    private function createProductUrl(PageInterface $page): SitemapUrlInterface
```
Comment (patrick477) [inline src/Sitemap/Provider/PageUrlProvider.php]: should this method be named `createPageUrl`?

File: src/Sitemap/Provider/PageUrlProvider.php [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+            ]);
+            if ($translation->getLocale() === $this->localeContext->getLocaleCode()) {
+                $pageUrl->setLocalization($location);
+                continue;
+            }
+            $pageUrl->addAlternative($location, $translation->getLocale());
+        }
+        return $pageUrl;
+    }
+
```
Comment (patrick477) [inline src/Sitemap/Provider/PageUrlProvider.php]: run the `bin/ecs check src --fix` command to correct the coding standard

File: src/Sitemap/Provider/PageUrlProvider.php (hunk 2 above)
Comment (sweoggy) [reply src/Sitemap/Provider/PageUrlProvider.php]: getLocale() can return null, this is just a safeguard against that.

**Comment by sweoggy** [comment]
Thanks @stefandoorn @patrick477 for your reviews, I have made changes accordingly 😃

File: composer.json [hunk 8]
```diff
@@ -19,15 +18,20 @@
         "friends-of-behat/service-container-extension": "^1.0",
         "friends-of-behat/symfony-extension": "^1.0",
         "friends-of-behat/variadic-extension": "^1.0",
+        "lakion/api-test-case": "^1.1",
```
Comment (stefandoorn) [inline composer.json]: I upgraded it to `^2.0|^3.1.0` last week, which went fairly easy.

**Comment by stefandoorn** [comment]
Looks good to me :)

File: composer.json (hunk 8 above)
Comment (sweoggy) [reply composer.json]: Surely we would need to upgrade to PHPUnit ^6.0 at least for the whole plugin then? Feels like that belongs to a separate PR.

Comment (stefandoorn) [reply composer.json]: True :)

**Comment by patrick477** [comment]
@sweoggy Thank you a lot! 🙂

**Pull request merged**